Description: Support polling 3
 polling 3 introduced the "events" structure as a replacement for
 a vec of events. It also introduced IO safety, which means that
 the poller.add function is now marked as unsafe. Afaict it's not any
 less safe than it was before, it's just marked differently now.
 
 rather than calling Events::new or Vec::new directly we make use
 of Default::default, this allows us to maintain compatibility
 with both the old and new versions of polling.
Author: Peter Michael Green <plugwash@debian.org>

Index: zbus-1/src/address.rs
===================================================================
--- zbus-1.orig/src/address.rs
+++ zbus-1/src/address.rs
@@ -31,8 +31,11 @@ impl Address {
                 let stream = unix(p)?;
 
                 let poller = Poller::new()?;
-                poller.add(&stream, Event::writable(0))?;
-                poller.wait(&mut Vec::new(), None)?;
+                unsafe {
+                    poller.add(&stream, Event::writable(0))?;
+                }
+                poller.wait(&mut Default::default(), None)?;
+                poller.delete(&stream); //the documentation says events must be removed from the poller before they are dropped.
 
                 stream.set_nonblocking(nonblocking)?;
 
Index: zbus-1/Cargo.toml
===================================================================
--- zbus-1.orig/Cargo.toml
+++ zbus-1/Cargo.toml
@@ -59,7 +59,7 @@ version = "0.26"
 version = "1.4.0"
 
 [dependencies.polling]
-version = "2.0.2"
+version = ">= 2.0.2"
 
 [dependencies.scoped-tls]
 version = "1.0.0"
